libctru/include/3ds/services/y2r.h

```diff
@@ -1,113 +1,361 @@
+/**
+ * @file y2r.h
+ */
 #pragma once
 #include <3ds/types.h>
 
+/**
+ * @brief Input color formats
+ *
+ * For the 16-bit per component formats, bits 31-16 are padding and 15-0 contains the value.
+ */
 typedef enum
 {
-	INPUT_YUV422_INDIV_8 = 0x0,
-	INPUT_YUV420_INDIV_8 = 0x1,
-	INPUT_YUV422_INDIV_16 = 0x2,
-	INPUT_YUV420_INDIV_16 = 0x3,
-	INPUT_YUV422_BATCH = 0x4,
+    INPUT_YUV422_INDIV_8  = 0x0, ///<  8-bit per component, planar YUV 4:2:2, 32bpp, (1 Cr & Cb sample per 2x1 Y samples).\n Usually named YUV422P.
+    INPUT_YUV420_INDIV_8  = 0x1, ///<  8-bit per component, planar YUV 4:2:0, 12bpp, (1 Cr & Cb sample per 2x2 Y samples).\n Usually named YUV420P.
+    INPUT_YUV422_INDIV_16 = 0x2, ///< 16-bit per component, planar YUV 4:2:2, 16bpp, (1 Cr & Cb sample per 2x1 Y samples).\n Usually named YUV422P16.
+    INPUT_YUV420_INDIV_16 = 0x3, ///< 16-bit per component, planar YUV 4:2:0, 24bpp, (1 Cr & Cb sample per 2x2 Y samples).\n Usually named YUV420P16.
+    INPUT_YUV422_BATCH    = 0x4, ///<  8-bit per component, packed YUV 4:2:2, 16bpp, (Y0 Cb Y1 Cr).\n Usually named YUYV422.
 } Y2R_InputFormat;
 
+/**
+ * @brief Output color formats
+ */
 typedef enum
 {
-	OUTPUT_RGB_32 = 0x0,
-	OUTPUT_RGB_24 = 0x1,
-	OUTPUT_RGB_16_555 = 0x2,
-	OUTPUT_RGB_16_565 = 0x3,
+    OUTPUT_RGB_32     = 0x0, ///< 32-bit, 0xRGBA order. A is the 8-bit value from @ref Y2RU_SetAlpha
+    OUTPUT_RGB_24     = 0x1, ///< 24-bit, 0xRGB order
+    OUTPUT_RGB_16_555 = 0x2, ///< 16-bit, 0bRRRRRGGGGGBBBBBA order. A is the 7th bit of the alpha value from @ref Y2RU_SetAlpha
+    OUTPUT_RGB_16_565 = 0x3, ///< 16-bit, 0bRRRRRGGGGGGBBBBB order
```

Comment:
I'd just say they match the framebuffer/texture formats and then add documentation for those in a better place. Y2R is relatively niche and it would be better if the color documentation was detailed with the framebuffer/gpu stuff.